### PR TITLE
campus interiors and great library faction

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Content "campus interiors and great library faction"

#### Purpose of change

Future faction base, build out final missing faction

#### Describe the solution

Introduce circulation and interiors like hallways, interior doors, elevators and stairs into the campus buildings, add to palette
Create barebones infrastructure for library faction and guards. Some guesses for where future bathrooms may go.

Guards have a tiny amount of dialogue and unique more civilian and DIY flavored gear.

#### Describe alternatives you've considered

Put the faction in another location like a wrecked evac shelter that is slowly fixed up and expanded. Dress all the guards in bookplate but that seems disrespectful to the books.

#### Testing

Game loads no errors. Map spawns and looks as expected.

#### Additional context

Thanks to the many who have worked on this

### CODE REVIEW

File: data/json/npcs/campus/generic_class_definitions.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+[
+  {
+    "type": "item_group",
+    "id": "FREE_MERCH_GUARD_wield_phase1",
```
Comment (MNG-cataclysm) [inline data/json/npcs/campus/generic_class_definitions.json:4]:
```suggestion
    "id": "CAMPUS_GUARD_wield_phase1",
```
The ID should probably be changed to match the faction, and that ID could cause conflict problems with Free Merchant NPCs.

File: data/json/npcs/factions.json [hunk 2]
```diff
@@ -944,5 +944,85 @@
...
+    "id": "the_great_library",
+    "name": "The Great Library",
+    "likes_u": 30,
+    "respects_u": 30,
+    "known_by_u": false,
+    "size": 100,
+    "power": 100,
+    "food_supply": 115200,
+    "wealth": 25000000,
+    "currency": "FMCNote",
```
Comment (MNG-cataclysm) [inline data/json/npcs/factions.json:959]:
```suggestion
    "currency": "FlatCoin",
```
There's an unused currency item meant for the faction, so they should probably use it.

File: data/json/npcs/campus/generic_class_definitions.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      { "item": "jacket_jean", "prob": 60 },
+      { "item": "jacket_leather", "prob": 20 },
+      { "item": "bandana", "prob": 10 }
+    ]
+  },
+  {
+    "type": "npc_class",
+    "id": "NC_CAMPUS_GUARD",
+    "name": { "str": "Guard" },
+    "job_description": "I'm on a guard shift.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/campus/generic_class_definitions.json:60]: The class should include the worn and wield overrides for their equipment and items.

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]

File: data/json/npcs/campus/generic_class_definitions.json (hunk 3 above)
Comment (LyleSY) [reply data/json/npcs/campus/generic_class_definitions.json:60]: Sigh, the original I was copying from was broken. Good eye, thanks!

Comment (MNG-cataclysm) [reply data/json/npcs/campus/generic_class_definitions.json:60]: Quite welcome.

**Pull request merged**